scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,196 @@
+
+        embedding_skill = AzureOpenAIEmbeddingSkill(
+            description="Skill to generate embeddings via Azure OpenAI",
+            context="/document/pages/*",
+            resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+            deployment_id=self.embeddings.open_ai_deployment,
+            inputs=[
+                InputFieldMappingEntry(name="text", source="/document/pages/*"),
+            ],
+            outputs=[OutputFieldMappingEntry(name="embedding", target_name="vector")],
+        )
+
+        index_projections = SearchIndexerIndexProjections(
+            selectors=[
+                SearchIndexerIndexProjectionSelector(
+                    target_index_name=index_name,
+                    parent_key_field_name="parent_id",
+                    source_context="/document/pages/*",
+                    mappings=[
+                        InputFieldMappingEntry(name="content", source="/document/pages/*"),
+                        InputFieldMappingEntry(name="embedding", source="/document/pages/*/vector"),
+                        InputFieldMappingEntry(name="sourcepage", source="/document/metadata_storage_name"),
+                    ],
+                ),
+            ],
+            parameters=SearchIndexerIndexProjectionsParameters(
+                projection_mode=IndexProjectionMode.SKIP_INDEXING_PARENT_DOCUMENTS
+            ),
+        )
+
+        skillset = SearchIndexerSkillset(
+            name=skillset_name,
+            description="Skillset to chunk documents and generate embeddings",
+            skills=[split_skill, embedding_skill],
+            index_projections=index_projections,
+        )
+
+        return skillset
+
+    async def setup(self, search_info: SearchInfo):
+        search_manager = SearchManager(
+            search_info=search_info,
+            search_analyzer_name=self.search_analyzer_name,
+            use_acls=self.use_acls,
+            use_int_vectorization=True,
+            embeddings=self.embeddings,
+            search_images=False,
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        await search_manager.create_index(
+            vectorizers=[
+                AzureOpenAIVectorizer(
+                    name=f"{search_info.index_name}-vectorizer",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=AzureOpenAIParameters(
+                        resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+                        deployment_id=self.embeddings.open_ai_deployment,
+                    ),
+                ),
+            ]
+        )
+
+        # create indexer client
+        ds_client = search_info.create_search_indexer_client()
+        ds_container = SearchIndexerDataContainer(name=self.blob_manager.container)
+        data_source_connection = SearchIndexerDataSourceConnection(
```

Comment:
Add deletion policy here, otherwise your removal of blobs below won't do much

Reply:
Good catch. I chose native blob soft delete policy. Please review the usage.